### PR TITLE
Fix incorrect behaviour of number validator in some locales

### PR DESCRIPTION
**Pull request by vlad-khramov**
In some locales (ru for example) decimal point is comma, so number validator didn't work properly in this locales, because string representation of floats in this locale contains comma instead of dot.

``` php
$val = 0.5;
print "$val"; // "0,5"
```

Build of commit with failing tests: https://travis-ci.org/quantum13/yii2/builds/99038650

### CODE REVIEW

File: tests/framework/validators/NumberValidatorTest.php [hunk 1]
```diff
@@ -37,7 +37,14 @@ public function testValidateValueSimple()
         $this->assertTrue($val->validate(-20));
         $this->assertTrue($val->validate('20'));
         $this->assertTrue($val->validate(25.45));
+
+        $oldLocale = setlocale(LC_ALL, "0");
+        setlocale(LC_ALL, "en_US.UTF-8");
```
Comment (samdark) [inline tests/framework/validators/NumberValidatorTest.php]: It won't run on Windows. Should be:

```php
setlocale(LC_ALL, 'en_US.UTF-8', 'English_United States.1252');
```

File: tests/framework/validators/NumberValidatorTest.php [hunk 2]
```diff
@@ -37,7 +37,14 @@ public function testValidateValueSimple()
...
         $this->assertFalse($val->validate('25,45'));
+        setlocale(LC_ALL, "en_DK.UTF-8"); //decimal point is comma
```
Comment (samdark) [inline tests/framework/validators/NumberValidatorTest.php]: It won't run on Windows. Should be:

```php
setlocale(LC_ALL, 'da_DK.UTF-8', 'Danish_Denmark.1252'); //decimal point is comma
```

File: framework/validators/NumberValidator.php [hunk 3]
```diff
@@ -117,6 +118,22 @@ protected function validateValue($value)
...
 
     /**
+     * Returns string represenation of number value with replaced commas to dots, if decimal point
+     * of current locale is comma
+     * @param $value
+     * @return string
+     */
+    private function getStringValue($value)
+    {
+        $localeInfo = localeconv();
```
Comment (samdark) [inline framework/validators/NumberValidator.php]: Could be something like:

```php
$value = (string)$value;

$locale = localeconv();
$decimalPointSeparator = isset($locale['decimal_point']) ? $locale['decimal_point'] : null;

if ($decimalPointSeparator !== null && $decimalPointSeparator !== '.') {
    $value = str_replace($decimalPointSeparator, '.', $value);
}

return $value;
```

**Comment by samdark** [comment]
It doesn't take locale's thousands separator into account.

**Comment by samdark** [comment]
@quantum13 do you have time to work further on it?

**Comment by thiagotalma** [comment]
This problem is VERY ANNOYING!

It would be great if this were solved ...

**Comment by samdark** [comment]
It is but also it's hard to solve properly one...

**Comment by samdark** [comment]
Closed in favor of https://github.com/yiisoft/yii2/issues/13114

**Pull request closed without merging**